doc/file-upload.md

```diff
@@ -0,0 +1,293 @@
+# Uploading files from an HTML form
+
+When uploading a file from an HTML form, the file content is not sent as the request body.
```

Comment:
nitpicky but this somehow contradicts itself - you mean the file content is not exclusively stored in the body but still part of the body right?

Reply:
I also do not really know what @nverwer meant with this. It is part of the request body, at least when sent with `multipart/form-data`.

Reply:
@nverwer any thoughts on this?